homeassistant/components/pglab/discovery.py

```diff
@@ -0,0 +1,277 @@
+        """Initialize the device discovery info."""
+
+        # hash string that represent the device actual configuration,
+        # it depends by the number of available relay and shutter
+        # When changed the device enties must be rebuilt
+        self._hash = pglab_device.hash
+        self._entities: list[tuple[str, str]] = []
+
+    def add_entity(self, entity: Entity) -> None:
+        """Add the entity id."""
+        if entity.unique_id:
+            self._entities.append((entity.platform.domain, entity.unique_id))
+
+    @property
+    def hash(self) -> int:
+        """Return the hash for this configuration."""
+        return self._hash
+
+    @property
+    def entities(self) -> list[tuple[str, str]]:
+        """Return array of entities available."""
+        return self._entities
+
+
+@dataclass
+class PGLabDiscovery:
+    """PG LAB Discovery class. Discovery a pglab/discovert/[device]/config."""
+
+    def __init__(self) -> None:
+        """Initialize the discovery class."""
+        self._substate: dict[str, EntitySubscription] = {}
+        self._discovery_topic = DISCOVERY_TOPIC
+        self._mqtt_client = None
+        self._discovered: dict[str, DiscoverDeviceInfo] = {}
+        self._disconnect_platform: list = []
+
+    async def __build_device(
+        self, mqtt: PyPGLabMqttClient, msg: ReceiveMessage
+    ) -> PyPGLabDevice:
+        """Build a PG LAB device."""
+
+        # check if the discovery message is in valid jason format
+        try:
+            payload = json.loads(msg.payload)
+        except ValueError:
+            _LOGGER.warning("Can't decode discovery payload: %s", msg.payload)
+            return None
+
+        device_id = "id"
+
+        # check the key id is present in the paylod. It must always present !!!
+        if device_id not in payload:
+            _LOGGER.warning("Unexpected discovery payload format, id key not present")
+            return None
+
+        # do a sanity check: the id must match the discovery topic /pglab/discovery/[id]/config
+        topic = msg.topic
+        if not topic.endswith(f"{payload[device_id]}/config"):
+            return None
+
+        # build and configure the PG LAB device
+        pglab_device = PyPGLabDevice()
+        if not await pglab_device.config(mqtt, payload):
+            _LOGGER.warning("Error during setup of new discovered device")
+            return None
+
+        return pglab_device
+
+    def __clean_discovered_device(self, hass: HomeAssistant, device_id: str) -> None:
+        """Destroy the device and any enties connected with the device."""
+
+        if device_id not in self._discovered:
+            return
+
+        discovery_info = self._discovered[device_id]
+
+        # destroy all entities connected with the device
+        entity_registry = er.async_get(hass)
+        for platform, entityid in discovery_info.entities:
+            entity_id = entity_registry.async_get_entity_id(platform, DOMAIN, entityid)
```

Comment:
This is not the entity_id, it's the unique_id, please change the variable names accordingly
```suggestion
        for platform, unique_id in discovery_info.entities:
            entity_id = entity_registry.async_get_entity_id(platform, DOMAIN, unique_id)
```

Reply:
I'll change it